test/plugin/test_filter_parser.rb

```diff
@@ -0,0 +1,730 @@
+require_relative '../helper'
+require 'timecop'
+require 'fluent/test/driver/filter'
+require 'fluent/plugin/filter_parser'
+require 'flexmock/test_unit'
+
+class ParserFilterTest < Test::Unit::TestCase
+  include FlexMock::TestCase
+
+  def setup
+    Fluent::Test.setup
+    @tag = 'test'
+    @default_time = Time.parse('2010-05-04 03:02:01 UTC')
+    Timecop.freeze(@default_time)
+  end
+
+  def teardown
+    super
+    Timecop.return
+  end
+
+  ParserError = Fluent::Plugin::Parser::ParserError
+  CONFIG = %[
+    key_name     message
+    reserve_data true
+    <parse>
+      @type regexp
+      expression /^(?<x>.)(?<y>.) (?<time>.+)$/
+      time_format %Y%m%d%H%M%S
+    </parse>
+  ]
+
+  def create_driver(conf=CONFIG)
+    Fluent::Test::Driver::Filter.new(Fluent::Plugin::ParserFilter).configure(conf)
+  end
+
+  def test_configure
+    assert_raise(Fluent::ConfigError) {
+      create_driver('')
+    }
+    assert_raise(Fluent::ConfigError) {
+      create_driver %[
+        key_name foo
+        <parse>
+          @type unknown_format_that_will_never_be_implemented
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name foo
+        <parse>
+          @type regexp
+          expression /(?<x>.)/
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name foo
+        <parse>
+          @type json
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name foo
+        format json
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name foo
+        <parse>
+          @type ltsv
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name message
+        suppress_parse_error_log true
+        <parse>
+          @type regexp
+          expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+        </parse>
+      ]
+    }
+    assert_nothing_raised {
+      create_driver %[
+        key_name message
+        suppress_parse_error_log false
+        <parse>
+          @type regexp
+          expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+        </parse>
+      ]
+    }
+    d = create_driver %[
+      key_name foo
+      <parse>
+        @type regexp
+        expression /(?<x>.)/
+      </parse>
+    ]
+    assert_false d.instance.reserve_data
+  end
+
+  # CONFIG = %[
+  #   remove_prefix test
+  #   add_prefix    parsed
+  #   key_name      message
+  #   format        /^(?<x>.)(?<y>.) (?<time>.+)$/
+  #   time_format   %Y%m%d%H%M%S
+  #   reserve_data  true
+  # ]
+  def test_filter
+    d1 = create_driver(CONFIG)
+    time = Time.parse("2012-01-02 13:14:15").to_i
```

Comment:
If this is a test code especially for integer time, test case name should show it.